packages/next/src/shared/lib/app-router-context.shared-runtime.ts

```diff
@@ -17,6 +17,11 @@ export type ChildSegmentMap = Map<string, CacheNode>
 export type CacheNode = ReadyCacheNode | LazyCacheNode
 
 export type LazyCacheNode = {
+  /**
+   * Whether the lazy cache node data promise has been resolved.
+   * This value is only true after we've called `use` on the promise (and applied the data to the tree).
+   */
+  lazyDataResolved?: boolean
```

Comment:
We shouldn't have optional fields because it breaks monomorphization and makes it slower. head is unfortunately already optional which we should also fix but we should at least not make it worse.

Reply:
but you can also do this in a follow up that fixes prefetchHead, head, and lazyDataResolved all at once.